### DISCUSSION TITLE
Using flox for daily climatology (getting 'kernel died')

### DISCUSSION THREAD

**Discussion by tlsw231** [pydata/xarray#7133] [Q&A] [answered]
I am trying to use `flox` to compute the climatology. I am guessing the way to do this is to just reduce along the appropriate label (day, month, etc.) using the `sum` function and then take the mean. The example below appears to almost go through on the dashboard but always ends up giving me a 'kernel died' message (My original dataset is about 21 times larger in size). Any ideas on what I might be doing wrong?
```
import xarray as xr

import flox
import flox.xarray
from flox.xarray import xarray_reduce

import dask
from dask.distributed import Client
from dask import delayed # to allow parallel computation
from distributed import Client
from dask_jobqueue import SLURMCluster

# Request cores
ncores=8; nprocs=ncores; totalcores= ncores * 1
cluster = SLURMCluster(cores=ncores, processes=nprocs, memory="264GB",queue="medium",
                       walltime="00:30:00")
cluster.scale(totalcores)
client = Client(cluster)

# Create dataarray of shape (1095, 28, 489, 655)
times = xr.cftime_range(start = '2011-01-01', end = '2013-12-31', freq='1D')
(nt,nz,ny,nx) =(len(times),28,489,655)
da = xr.DataArray(data=dask.array.random.random([nt,nz,ny,nx]),
                  dims=['time','z','y','x'],coords=dict(time=times)).rename('dummy')

# Create additional label to be used with xarray_reduce
da['time_dayofyear']=da.time.dt.dayofyear

# Reduce by computing sum along 'dayofyear' labels
da_clim = xarray_reduce(da, da['time_dayofyear'], method="map-reduce",func="sum").compute()
```
![pic1](https://user-images.githubusercontent.com/90293171/194134961-2304fa4a-aac2-4cd3-923e-cacb74f14589.png)

![pic2](https://user-images.githubusercontent.com/90293171/194135012-f615dc17-5193-4497-a6da-ff08b047580f.png)

**Comment by tlsw231** [comment, marked as answer]
I am marking this as answered because I ended up asking the wrong question (!) but am unable to delete the discussion. Will start another thread later which is about `flox` and resampling.

**Reply by dcherian** [reply]
For the groupby problem, there's some discussion here:https://flox.readthedocs.io/en/latest/user-stories/climatology.html
